Classes/PKStateMachine.swift

```diff
@@ -0,0 +1,72 @@
+//
+//  PKStateMachine.swift
+//  Pods
+//
+//  Created by Gal Orlanczyk on 02/04/2017.
+//
+//
+
+import Foundation
+
+protocol IntRawRepresentable: RawRepresentable {
+    var rawValue: Int { get }
+}
+
+protocol StateProtocol: IntRawRepresentable, Hashable {}
+
+extension StateProtocol {
+    var hashValue: Int {
+        return rawValue
+    }
+}
+
+class BasicStateMachine<T: StateProtocol> {
```

Comment:
I want it to be easy to use this state machine in future if it will be needed. and yes I am using it in the second pull I will push after this one in the ads player decorator. I had more features and more advance state machine implementation (with transition table and more) but kept it simple.